src/hotspot/share/logging/logTagSet.cpp

```diff
@@ -126,12 +126,17 @@ void LogTagSet::vwrite(LogLevelType level, const char* fmt, va_list args) {
   assert(ret >= 0, "Log message buffer issue");
   if ((size_t)ret >= sizeof(buf)) {
     size_t newbuf_len = prefix_len + ret + 1;
-    char* newbuf = NEW_C_HEAP_ARRAY(char, newbuf_len, mtLogging);
+    char* newbuf = (char*)::calloc(newbuf_len, sizeof(char));
```

Comment:
Can you precede this with a little comment please: 
`// We use raw ::malloc to prevent circularities`